### PR TITLE
Feature/413 docker proxy

### LINKED ISSUE #413: Proxy server for docker images
There is an existing mechanism to use a proxy server when pulling dogu information from dogu registry. This proxy does not work for the docker registry. This should be possible.

### PR DESCRIPTION
**Pull request by sklein94**
Resolve #413

### CODE REVIEW

File: resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  etcdctl get config/_global/proxy/enabled || echo "false"
+}
+
+if [ "$(get_enabled)" == "true" ]
+then
+  echo "collecting https proxy config steps..."
+  HOST="$(etcdctl get config/_global/proxy/server)" || (echo "ERROR: Could not read from etcd: proxy host" && exit 0)
+  PORT="$(etcdctl get config/_global/proxy/port)" || (echo "ERROR: Could not read from etcd: proxy port" && exit 0)
+  USERNAME="$(etcdctl get config/_global/proxy/username)" || echo "WARNING: Could not read from etcd: proxy host"
+  PASSWORD="$(etcdctl get config/_global/proxy/password)" || echo "WARNING: Could not read from etcd: proxy host"
```
Comment (pistengeier) [inline resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh:15]:
```suggestion
  USERNAME="$(etcdctl get config/_global/proxy/username)" || echo "WARNING: Could not read from etcd: proxy username"
  PASSWORD="$(etcdctl get config/_global/proxy/password)" || echo "WARNING: Could not read from etcd: proxy password"
```

File: docs/docs/user-guide/setup-json.md [hunk 2]
```diff
@@ -400,6 +399,39 @@ The following objects are examples:
 Hostnames must not occur multiple times.
 It is also not possible to add the same hostnames as in global configuration in the cesapp configuration.
 
+##### Section `config/_global/proxy`
+
+Use this section to configure a proxy server which currently is used to access the dogu registry, the docker registry.
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json.md:404]:
```suggestion
Use this section to configure a proxy server which currently is used to access the dogu and docker registry.
```

File: docs/docs/user-guide/setup-json_de.md [hunk 3]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
 Hostnamen dürfen nicht mehrfach vorkommen. 
 Außerdem darf derselbe Hostname nicht in der Konfiguration der Cesapp und in der globalen Konfiguration konfiguriert sein.
 
+##### Section `config/_global/proxy`
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:408]:
```suggestion
##### Bereich `config/_global/proxy`
```

File: docs/docs/user-guide/setup-json_de.md [hunk 4]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
...
+
+Hier kann ein Proxy-Server konfiguriert werden, der unter anderem für die Kommunikation mit der Dogu-Registry 
+und der Docker-Registry verwendet wird.
+Hinweis: Der Docker-Service muss neu gestartet werden, damit die Änderungen Anwendung finden.
+
+###### `server`
+
+* Datentyp: string
+* Inhalt: The ip address or hostname of the server (z.B `192.168.56.2` oder `www.example.com`)
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:417]: Please use the german translation.

File: docs/docs/user-guide/setup-json_de.md [hunk 5]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
...
+
+###### `port`
+
+* Datentyp: string
+* Inhalt: The port of the server (z.B `3128`)
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:422]: Please use the german translation.

File: docs/docs/user-guide/setup-json_de.md [hunk 6]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
...
+
+###### `username`
+
+* Optional
+* Datentyp: string
+* Inhalt: The username which is used to authenticate with the server
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:428]: Please use the german translation.

File: docs/docs/user-guide/setup-json_de.md [hunk 7]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
...
+
+###### `password`
+
+* Optional
+* Datentyp: string
+* Inhalt: The password which is used to authenticate with the server
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:434]: Please use the german translation.

File: docs/docs/user-guide/setup-json_de.md [hunk 8]
```diff
@@ -405,6 +405,39 @@ Enthält beispielsweise folgende Objekte:
...
+
+###### `no_proxy`
+
+* Optional
+* Datentyp: string
+* Inhalt: Comma separated list of hosts which can be reached without the proxy server (e.g. `*.test.example.com,.example2.com'`).
```
Comment (pistengeier) [inline docs/docs/user-guide/setup-json_de.md:440]: Please use the german translation.

File: resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  AUTH=""
+  if [ "${USERNAME}" != "" ] && [ "${USERNAME}" != "" ]
+  then
+    AUTH="${USERNAME}:${PASSWORD}@"
+  else
+    echo "No username and password for proxy configured."
+  fi
+
+  HTTP_CONFIG="HTTP_PROXY=http://${HOST}:${PORT}"
```
Comment (pistengeier) [inline resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh:24]: Is there no authentication required for `HTTP_PROXY`?

File: setup.json [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+          "port": "3128",
+          "authentication": {
+            "username": "admin",
+            "password": "adminpw"
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (pistengeier) [inline setup.json:53]: Remove this file from the repository.

File: resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh (hunk 1 above)
Comment (sklein94) [reply resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh]: This code has been removed

File: resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh (hunk 9 above)
Comment (sklein94) [reply resources/etc/systemd/system/docker-metadata.service.d/docker-start-script.sh]: This code has been removed.

File: setup.json (hunk 10 above)
Comment (sklein94) [reply setup.json]: This code has been removed.

**Review by pistengeier — APPROVED** [review]

**Pull request merged**